mail_debrand/models/mail_render_mixin.py

```diff
@@ -12,43 +12,61 @@
 class MailRenderMixin(models.AbstractModel):
     _inherit = "mail.render.mixin"
 
-    def remove_href_odoo(self, value, remove_parent=True, remove_before=False):
+    def remove_href_odoo(
+        self, value, remove_parent=True, remove_before=False, to_keep=None
+    ):
         if len(value) < 20:
             return value
         # value can be bytes type; ensure we get a proper string
         if type(value) is bytes:
             value = value.decode()
         has_odoo_link = re.search(r"<a\s(.*)odoo\.com", value, flags=re.IGNORECASE)
         if has_odoo_link:
-            tree = etree.HTML(
-                value
-            )  # html with broken links   tree = etree.fromstring(value) just xml
-            odoo_achors = tree.xpath('//a[contains(@href,"odoo.com")]')
-            for elem in odoo_achors:
-                parent = elem.getparent()
-                previous = elem.getprevious()
+            # We don't want to change what was explicitly added in the message body,
+            # so we will only change what is before and after it.
+            if to_keep:
+                to_change = value.split(to_keep)
+            else:
+                to_change = [value]
+                to_keep = ""
+            new_parts = []
+            for part in to_change:
+                tree = etree.HTML(
```

Comment:
As you are processing parts, possibly this is better suited:

```suggestion
                tree = html.fromstring(
```